### PR TITLE
feat: Add goal tracking: delete all entries, closes #303

### LINKED ISSUE #303: Add goal tracking to delete all entries feature
See instruction in #302

### PR DESCRIPTION
**Pull request by olavea**
When the user clicks the delete all entries button it should be tracked. (Because bleh, bleh, bleh)

Pretend FATHOM_EXPORT = FATHOM_DELETE_ALL_ENTRIES

1. Add GATSBY_FATHOM_EXPORT to your .env.development (any value will due)
2. Export FATHOM_EXPORT from `./constants`
3. Import the `trackGoal` function from the `tracking` feature. 
4. Import `FATHOM_EXPORT` from `./constants`
5. When button is clicked call `trackGoal` with `FATHOM_EXPORT` as param
6. Test and make sure the console logs out `Track fathom goal <FATHOM_EXPORT>)`
7. add a `FATHOM_DELETE_CONFIRMED` as well

@raae will add a real value to Netlify env variables so this will be tracked properly by fathom in production.

### CODE REVIEW

File: src/features/profile/DangerCard.js [hunk 1]
```diff
@@ -39,12 +39,12 @@ const DangerCard = () => {
...
     })
   }, [entries, mensesTags, isPending])
-
+  const trackGoal = (FATHOM_EXPORT) => {}
   const handleDeleteAllEntries = async () => {
     const CONFIRMATION_STRING = "DELETE"
 
     setIsPending(true)
-
+    //here
```
Comment (raae) [inline src/features/profile/DangerCard.js:47]: Correct

File: src/features/profile/DangerCard.js [hunk 2]
```diff
@@ -39,12 +39,12 @@ const DangerCard = () => {
...
     const confirmation = prompt(
       `Please type ${CONFIRMATION_STRING} to confirm deletion of all your entries.`
     )
```
Comment (raae) [inline src/features/profile/DangerCard.js:50]: Maybe we should add a FATHOM_DELETE_CONFIRMED as well?

**Review by raae — CHANGES_REQUESTED** [review]

File: src/features/profile/DangerCard.js (hunk 2 above)
Comment (olavea) [reply src/features/profile/DangerCard.js]: 👍

**Comment by olavea** [comment]
Can I merge this now Queen @raae 👑 ?

File: src/features/profile/DangerCard.js [hunk 3]
```diff
@@ -64,6 +70,8 @@ const DangerCard = () => {
         alert(`Success, all your entries were deleted.`)
       }
     }
+    // call trackGoal here, when the user succeeds in deleting entries
+    trackGoal(FATHOM_DELETE_CONFIRMED);
```
Comment (raae) [inline src/features/profile/DangerCard.js:74]: **suggestion:** This needs to be moved into the else above, so it is only triggered when it's successful.

**Review by raae — CHANGES_REQUESTED** [review]

File: src/features/profile/DangerCard.js [hunk 4]
```diff
@@ -15,6 +15,10 @@ import { CardContentSection } from "../../components"
 import { selectAllEntries, deleteAllEntries } from "../entries"
 import { useSettings } from "../settings"
 
+import { FATHOM_DELETE_ALL_ENTRIES, FATHOM_DELETE_CONFIRMED } from "../../constants";
+import {trackGoal} from "../tracking"
```
Comment (raae) [inline src/features/profile/DangerCard.js:19]: **thought:** Looks like prettier is not working properly. Should be a space before and after `trackGoal`

File: src/features/profile/DangerCard.js (hunk 3 above)
Comment (olavea) [reply src/features/profile/DangerCard.js]: so it is only triggered when it's successful #303
e6a4025

File: src/features/profile/DangerCard.js (hunk 4 above)
Comment (olavea) [reply src/features/profile/DangerCard.js]: Should be a space before and after trackGoal #303
b2d9f3b

File: src/features/profile/DangerCard.js [hunk 5]
```diff
@@ -68,10 +68,10 @@ const DangerCard = () => {
         alert(`Oopsie (${error.message}), please try again.`)
       } else {
         alert(`Success, all your entries were deleted.`)
+        // call trackGoal here, when the customer has succeeded in deleting her/his/their entries
+        trackGoal(FATHOM_DELETE_CONFIRMED);
```
Comment (raae) [inline src/features/profile/DangerCard.js:72]: **suggestion:** Move above the alert, in case user does not click inside alert and just closes the window.

**Review by raae — CHANGES_REQUESTED** [review]

File: src/features/profile/DangerCard.js (hunk 5 above)
Comment (olavea) [reply src/features/profile/DangerCard.js:71]: if customer: not click alert & closes window #303
bf3c926

File: src/features/profile/DangerCard.js [hunk 6]
```diff
@@ -61,7 +67,10 @@ const DangerCard = () => {
       if (error) {
         alert(`Oopsie (${error.message}), please try again.`)
       } else {
+        // call trackGoal here, when the customer has succeeded in deleting her/his/their entries
+        trackGoal(FATHOM_DELETE_CONFIRMED);
```
Comment (raae) [inline src/features/profile/DangerCard.js:71]: **suggestion:** Did not see this until now, but has to be FATHOM_DELETE_ALL_ENTRIES_CONFIRMED

Comment (raae) [reply src/features/profile/DangerCard.js:71]: I have added it to Netlify as GATSBY_FATHOM_DELETE_ALL_ENTRIES_CONFIRMED, so should show in console.log as soon as this is changed and the PR builds again.

**Review by raae — CHANGES_REQUESTED** [review]

File: src/features/profile/DangerCard.js (hunk 6 above)
Comment (olavea) [reply src/features/profile/DangerCard.js]: FATHOM_DELETE_ALL_ENTRIES_CONFIRMED #303
942cc32

Comment (olavea) [reply src/features/profile/DangerCard.js]: FATHOM_DELETE_ALL_ENTRIES_CONFIRMED #303
05b6b9a

**Review by raae — APPROVED** [review]

**Comment by olavea** [comment]
365DaysOfPOWprogramming-day-156_feat- Add goal tracking- delete all entries, closes #303 #306 -374

**Pull request merged**